### PR TITLE
Balamcan/feature/google oauth

### PR DESCRIPTION
**Pull request by balamcapital**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please select option.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested (manually)?

- [x] Create User model
- [x] Create User Credential model

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: www/.gitignore
Comment (alfonsocv12) [inline www/.gitignore:1]: I think we don't need this one

File: www/api/libs/utils/authorize/authorize.py [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 from api.libs.domain_entity import UserType
+from shared.tipsy_jwt import verify_token
+from functools import wraps
 
 
 def authorize(user_type_required: UserType):
+	def login_required(f):
+		@wraps(f)
+		def decorated_function(*args, **kwargs):
+			# authorization_token = request.headers.get('Authorization', None)
```
Comment (alfonsocv12) [inline www/api/libs/utils/authorize/authorize.py]: Is this comment needed?

File: www/api/libs/utils/authorize/authorize.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+			token = request.headers.get('Authorization').split()[1]
```
Comment (alfonsocv12) [inline www/api/libs/utils/authorize/authorize.py]: its there a way to avoid using magic numbers?

File: www/api/libs/utils/authorize/authorize.py [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
-	# 		if g.user is None:
-	# 			return redirect(url_for('login', next=request.url))
-	# 		return f(*args, **kwargs)
-	# 	return decorated_function
-	
-	# return login_required
\ No newline at end of file
+			if verification_result is None:
+				return redirect(url_for('login', next=request.url))
+			return f(*args, **kwargs)
```
Comment (alfonsocv12) [inline www/api/libs/utils/authorize/authorize.py]: Its interesting the different return types

File: www/api/modules/user/domain/service/user_service.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  def get_user_by_id(self, user_id: str) -> UserModel:
+    # Get user from the repository by id
+    user = self.user_repository.get_user_by_id(user_id)
+    return user
+
+  def get_user_by_email(self, email: str) -> UserModel:
+    # Get user from the repository by email
+    user = self.user_repository.get_user_by_id(email)
+    return user
```
Comment (alfonsocv12) [inline www/api/modules/user/domain/service/user_service.py]: NIT why not `return self.user_repository...`

File: www/api/modules/user/domain/service/user_service.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    
+    user = self.user_repository.create_user(user)
+
+    return None, user
+
+  def update_user(self, user_id, updated_data):
+    # Update user in the repository
+    user = self.user_repository.update_user(user_id, updated_data)
+
+    return user
```
Comment (alfonsocv12) [inline www/api/modules/user/domain/service/user_service.py]: same here

File: www/api/modules/user/domain/service/user_service.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  def delete_user(self, user_id):
+    # Delete user from the repository
+    self.user_repository.delete_user(user_id)
```
Comment (alfonsocv12) [inline www/api/modules/user/domain/service/user_service.py]: What happends if there's and error while deleting the user?

File: www/app.py [hunk 7]
```diff
@@ -14,6 +15,10 @@ def create_app():
     def health():
         return 'OK'
 
+    flask_app.secret_key = SECRET_KEY
+    # TODO: Not sure if this is the best session type
```
Comment (alfonsocv12) [inline www/app.py]: Did you had time to solve this?

File: www/shared/globals.py [hunk 8]
```diff
@@ -31,3 +36,7 @@
 }
 
 local_prefix = os.getenv('LOCAL_PREFIX', './tmp')
+
+SECRET_KEY = os.getenv('SECRET_KEY', 'Super Scret Key For Flask App')
```
Comment (alfonsocv12) [inline www/shared/globals.py:39]: I think all of this can be manage by `jwt_settings` to continue with the combention

File: www/app.py (hunk 7 above)
Comment (balamcapital) [reply www/app.py]: i got some doubts to change this...

**Review by alfonsocv12 — APPROVED** [review]

**Review by alfonsocv12 — APPROVED** [review]

**Pull request merged**